grakn-core/src/main/java/ai/grakn/QueryRunner.java

```diff
@@ -0,0 +1,91 @@
+/*
+ * Grakn - A Distributed Semantic Database
+ * Copyright (C) 2016-2018 Grakn Labs Limited
+ *
+ * Grakn is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Grakn is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Grakn. If not, see <http://www.gnu.org/licenses/gpl.txt>.
+ */
+
+package ai.grakn;
+
+import ai.grakn.concept.Concept;
+import ai.grakn.graql.AggregateQuery;
+import ai.grakn.graql.DefineQuery;
+import ai.grakn.graql.DeleteQuery;
+import ai.grakn.graql.GetQuery;
+import ai.grakn.graql.InsertQuery;
+import ai.grakn.graql.UndefineQuery;
+import ai.grakn.graql.admin.Answer;
+import ai.grakn.graql.analytics.ClusterQuery;
+import ai.grakn.graql.analytics.CorenessQuery;
+import ai.grakn.graql.analytics.CountQuery;
+import ai.grakn.graql.analytics.DegreeQuery;
+import ai.grakn.graql.analytics.KCoreQuery;
+import ai.grakn.graql.analytics.MaxQuery;
+import ai.grakn.graql.analytics.MeanQuery;
+import ai.grakn.graql.analytics.MedianQuery;
+import ai.grakn.graql.analytics.MinQuery;
+import ai.grakn.graql.analytics.PathQuery;
+import ai.grakn.graql.analytics.PathsQuery;
+import ai.grakn.graql.analytics.StdQuery;
+import ai.grakn.graql.analytics.SumQuery;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+import java.util.stream.Stream;
+
+/**
+ * @author Felix Chapman
+ */
+public interface QueryRunner {
```

Comment:
I think you might want to update the java docs to say why we need this interface. I.e. So we can decouple the queries and their execution,

Reply:
good point!